File: src/game/state/StateManager.cpp
//
// Created by Per-Arne on 24.02.2017.
//

#include "stateManager.h"

StateManager::StateManager()
{
	walkingState = std::shared_ptr<BaseState>(new Walking());
	spawnState = std::shared_ptr<BaseState>(new Spawning());
	idleState = std::shared_ptr<BaseState>(new Idle());
	despawnedState = std::shared_ptr<BaseState>(new Despawned());
	harvestingState = std::shared_ptr<BaseState>(new Harvesting());
	buildingState = std::shared_ptr<BaseState>(new Building());
	combatState = std::shared_ptr<BaseState>(new Combat());
	deadState = std::shared_ptr<BaseState>(new Dead());

}


std::shared_ptr<BaseState> StateManager::getByID(int id){
    switch(id) {
        case Constants::State::Building:
            return buildingState;
        case Constants::State::Spawning:
            return spawnState;
        case Constants::State::Walking:
            return walkingState;
        case Constants::State::Idle:
            return idleState;
        case Constants::State::Despawned:
            return despawnedState;
        case Constants::State::Harvesting:
            return harvestingState;
        case Constants::State::Combat:
            return combatState;
        case Constants::State::Dead:
            return deadState;
    }
}

File: python/game/state/StateManager.py
from game import Constants
from game import state


class StateManager:

    walking_state = state.Building()
    spawn_state = state.Spawning()
    idle_state = state.Idle()
    despawned_state = state.Despawned()
    harvesting_state = state.Harvesting()
    building_state = state.Building()
    combat_state = state.Combat()
    dead_state = state.Dead()

    @staticmethod
    def get_by_id(state_type_id):
        if state_type_id == Constants.State.Building:
            return StateManager.walking_state
        elif state_type_id == Constants.State.Spawning:
            return StateManager.spawn_state
        elif state_type_id == Constants.State.Walking:
            return StateManager.walking_state
        elif state_type_id == Constants.State.Idle:
            return StateManager.idle_state
        elif state_type_id == Constants.State.Despawned:
            return StateManager.despawned_state
        elif state_type_id == Constants.State.Harvesting:
            return StateManager.harvesting_state
        elif state_type_id == Constants.State.Combat:
            return StateManager.combat_state
        elif state_type_id == Constants.State.Dead:
            return StateManager.dead_state
        else:
            assert False  # Should NEVER happen